twitter.go

```diff
@@ -273,3 +279,26 @@ func (c *Client) handleMediaUpload(ctx context.Context, method, urlStr string, q
 		Media:     mediaUpload,
 	}, nil
 }
+
+func (c *Client) handleGetInsights(ctx context.Context, method, urlStr string, query getInsightsQueryResponse) (*PostInsightsResponse, error) {
+	resp, err := c.execute(ctx, method, urlStr, query.ContentType, query.Body, nil)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+
+	var insightsResp MediaInsights
+	err = json.NewDecoder(resp.Body).Decode(&insightsResp)
+	if err != nil && err != io.EOF {
```

Comment:
I think `io.EOF` for this case is not necessary!

Reply:
Thank you!